cmd/osm-controller/osm-controller_test.go

```diff
@@ -2,74 +2,202 @@ package main
+		initialDebugServerEnabled  bool
+		changeDebugServerEnabledTo bool
+		c                          controller
+		expectedStopCount          int
+		expectedStopErr            bool
+		expectedDebugServerRunning bool
+	}{
+		{
+			name:                       "turn off debug server",
+			initialDebugServerEnabled:  true,
+			changeDebugServerEnabledTo: false,
+			c:                          controller{debugServerRunning: true, debugComponents: mockDebugConfig, debugServer: &fakeDebugServer},
+			expectedStopCount:          1,
+			expectedStopErr:            false,
+			expectedDebugServerRunning: false,
+		},
+		{
+			name:                       "turn on debug server",
+			initialDebugServerEnabled:  false,
+			changeDebugServerEnabledTo: true,
+			c:                          controller{debugServerRunning: false, debugComponents: mockDebugConfig, debugServer: nil},
+			expectedStopCount:          0,
+			expectedStopErr:            false,
+			expectedDebugServerRunning: true,
+		},
+		{
+			name:                       "error when turning off debug server",
+			initialDebugServerEnabled:  true,
+			changeDebugServerEnabledTo: false,
+			c:                          controller{debugServerRunning: true, debugComponents: mockDebugConfig, debugServer: &fakeDebugServerGetErr},
+			expectedStopCount:          1,
+			expectedStopErr:            true,
+			expectedDebugServerRunning: false,
+		},
+		{
+			name:                       "debug server is already on, do nothing",
+			initialDebugServerEnabled:  true,
+			changeDebugServerEnabledTo: true,
+			c:                          controller{debugServerRunning: true, debugComponents: mockDebugConfig, debugServer: nil},
+			expectedStopCount:          0,
+			expectedStopErr:            false,
+			expectedDebugServerRunning: true,
+		},
+		{
+			name:                       "debug server is already off, do nothing",
+			initialDebugServerEnabled:  false,
+			changeDebugServerEnabledTo: false,
+			c:                          controller{debugServerRunning: false, debugComponents: mockDebugConfig, debugServer: nil},
+			expectedStopCount:          0,
+			expectedStopErr:            false,
+			expectedDebugServerRunning: false,
+		},
+	}
+
+	defaultConfigMap := map[string]string{}
+	configMap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: osmNamespace,
+			Name:      osmConfigMapName,
+		},
+		Data: defaultConfigMap,
+	}
+	_, err := kubeClient.CoreV1().ConfigMaps(osmNamespace).Create(context.TODO(), &configMap, metav1.CreateOptions{})
+	assert.Nil(err)
+
+	for _, tests := range testCases {
+		t.Run(fmt.Sprintf("Test: %s", tests.name), func(t *testing.T) {
+			defaultConfigMap["enable_debug_server"] = strconv.FormatBool(tests.changeDebugServerEnabledTo)
+			configMap.Data = defaultConfigMap
+
+			go tests.c.configureDebugServer(cfg)
+			_, err := kubeClient.CoreV1().ConfigMaps(osmNamespace).Update(context.TODO(), &configMap, metav1.UpdateOptions{})
+			assert.Nil(err)
+			time.Sleep(time.Second)
```

Comment:
good rule of thumb is to never have a sleep in a unit test. This makes tests flakey and brittle in the long run.